libbeat/kibana/transform.go

```diff
@@ -7,113 +7,170 @@ import (
+			if f.Enabled == nil || *f.Enabled {
+				t.transformFields(f.Fields, f.Path)
+			}
 		} else {
-			// set default values (as done in python script)
-			keys[f.Path] = true
-
-			truthy := true
-			falsy := false
-			f.Index = &truthy
-			f.Analyzed = &falsy
-			f.DocValues = &truthy
-			f.Searchable = &truthy
-			f.Aggregatable = &truthy
-			add(f, fields, fieldFormatMap)
+			t.keys[f.Path] = true
+			t.add(f)
+			if f.MultiFields != nil {
+				path := f.Path
+				for _, mf := range f.MultiFields {
+					f.Type = mf.Type
+					f.Path = path + "." + mf.Name
+					t.add(f)
+				}
+			}
 		}
 	}
 }
 
-func add(f common.Field, fields *[]common.MapStr, fieldFormatMap common.MapStr) {
+func (t *Transformer) add(f common.Field) {
 	field, fieldFormat := transformField(f)
-	*fields = append(*fields, field)
+	t.transformedFields = append(t.transformedFields, field)
 	if fieldFormat != nil {
-		fieldFormatMap[field["name"].(string)] = fieldFormat
+		t.transformedFieldFormatMap[field["name"].(string)] = fieldFormat
 	}
 
 }
 
 func transformField(f common.Field) (common.MapStr, common.MapStr) {
 	field := common.MapStr{
-		"name":         f.Path,
-		"count":        0,
-		"scripted":     false,
-		"indexed":      getVal(f.Index, true),
-		"analyzed":     getVal(f.Analyzed, false),
-		"doc_values":   getVal(f.DocValues, true),
-		"searchable":   getVal(f.Searchable, true),
-		"aggregatable": getVal(f.Aggregatable, true),
+		"name":       f.Path,
+		"type":       typeMapping[f.Type],
+		"count":      f.Count,
+		"scripted":   false,
+		"indexed":    getVal(f.Index, true),
+		"searchable": getVal(f.Searchable, true),
 	}
 
-	if t, ok := typeMapping[f.Type]; ok == true {
-		field["type"] = t
+	setReadFromDocValues(field, f)
+	setAggregatable(field, f)
+
+	if f.Script != "" {
+		field["scripted"] = true
+		field["script"] = f.Script
+		field["indexed"] = false
+		if f.Lang == "" {
+			field["lang"] = "painless"
+		} else {
+			field["lang"] = f.Lang
+		}
 	}
 
-	if f.Type == "text" {
+	if field["indexed"] == false {
+		field["searchable"] = false
 		field["aggregatable"] = false
+		field["readFromDocValues"] = false
 	}
 
 	var format common.MapStr
-	if f.Format != "" || f.Pattern != "" {
-		format = common.MapStr{}
-
-		if f.Format != "" {
-			format["id"] = f.Format
-			if f.InputFormat != "" {
-				format["params"] = common.MapStr{"inputFormat": f.InputFormat}
-			}
-		}
-		if f.Pattern != "" {
-			format["params"] = common.MapStr{"pattern": f.Pattern}
-		}
+	if f.Format != "" {
+		format = common.MapStr{"id": f.Format}
 	}
+	addFormatParam(format, "pattern", f.Pattern)
+	addFormatParam(format, "inputFormat", f.InputFormat)
+	addFormatParam(format, "outputFormat", f.OutputFormat)
+	addFormatParam(format, "outputPrecision", f.OutputPrecision)
+	addFormatParam(format, "labelTemplate", f.LabelTemplate)
+	addFormatParam(format, "urlTemplate", f.UrlTemplate)
 
 	return field, format
 }
 
+func setReadFromDocValues(f common.MapStr, c common.Field) {
```

Comment:
@simianhacker should it be `doc_values` or `readFromDocValues` for the Kibana index pattern?